src/field.rs

```diff
@@ -886,15 +884,37 @@ pub(crate) fn split_vector<F: FieldElement>(inp: &[F], num_shares: usize) -> Vec
 
     for _ in 1..num_shares {
         let share: Vec<F> = random_vector(inp.len());
-        for (x, y) in outp[0].iter_mut().zip(&share) {
-            *x -= *y;
-        }
+        sub_assign_vector(&mut outp[0], share.iter().copied());
         outp.push(share);
     }
 
     outp
 }
 
+pub(crate) fn sub_assign_vector<F: FieldElement>(a: &mut [F], b: impl IntoIterator<Item = F>) {
+    let mut count = 0;
+    for (x, y) in a.iter_mut().zip(b) {
+        *x -= y;
+        count += 1;
+    }
+    assert_eq!(a.len(), count);
```

Comment:
If we wanted to be real clever, we could add a debug assertion that a.len == b.len using `#[cfg(debug_assertions)]` to conditionally compile it.

Reply:
At this point, we only know `b` produces an iterator, so we'd have to do so in a more specialized method that takes two slices